src/app/core/uri-retriever.service.ts

```diff
@@ -0,0 +1,245 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Observable, throwError, Subject } from 'rxjs';
+import { LearningObject } from '@entity';
+import { USER_ROUTES, PUBLIC_LEARNING_OBJECT_ROUTES } from '@env/route';
+import { LearningOutcome } from '@entity';
+import { catchError, retry, map, tap, filter, take, takeUntil, finalize } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UriRetrieverService {
+
+  constructor(private http: HttpClient) { }
+
+  /**
+   * This function will return the metadata and the requested resources for a learning object as a promise of a Learning Object.
+   * @params params{ author: string, name: string, id:string } the values needed to retrieve the metadata for a learning object
+   * @params resources the resources (i.e. children, parents, outcomes, etc) that need to be loaded with the metadata
+   *
+   *
+   */
+  async getLearningObject(
+    params: { author?: string, name?: string, id?: string },
+    resources?: string[]
+  ): Promise<LearningObject> {
+    let properties: string[];
+    if (resources) {
+      properties = resources;
+    } else {
+      properties = [];
+    }
+
+    const learningObject = {};
+    const request = this.getLearningObjectResources(params, properties);
+
+    return new Promise((resolve) => {
```

Comment:
There is a lot happening here. Can you abstract this to a helper function to improve readability?